test/e2e/query_offset.test.ts

```diff
@@ -0,0 +1,65 @@
+import { expect } from "chai"
+import { Connection } from "../../src"
+import { Message } from "../../src/producer"
+import { Offset } from "../../src/requests/subscribe_request"
+import { Rabbit } from "../support/rabbit"
+import { expectToThrowAsync, password, username } from "../support/util"
+import { createConnection } from "../support/fake_data"
+
+describe("declare consumer", () => {
+  const rabbit = new Rabbit(username, password)
+  const testStreamName = "test-stream"
+  let connection: Connection
+
+  beforeEach(async () => {
+    await rabbit.createStream(testStreamName)
+    connection = await createConnection(username, password)
+  })
+
+  afterEach(async () => {
+    await connection.close()
+    try {
+      await rabbit.deleteStream(testStreamName)
+    } catch (e) {}
+  })
+
+  it("the consumer is able to track the offset of the stream through queryOffset method", async () => {
+    const messages: Buffer[] = []
+    const publisher = await connection.declarePublisher({ stream: testStreamName })
+    await publisher.send(Buffer.from("hello"))
+    const consumer = await connection.declareConsumer(
+      { stream: testStreamName, offset: Offset.first(), consumerRef: "my_consumer" },
+      (message: Message) => messages.push(message.content)
+    )
+    await consumer.storeOffset(1n)
```

Comment:
Move the consumer inside the consumer

Reply:
Sorry, you want me to move the storeOffset where?